master/buildbot/db/builds.py

```
@@ -276,6 +276,84 @@ def thd(conn):
+                    'branch': branch,
+                    'revision': revision,
+                    'patchid': None,
+                    'repository': repository,
+                    'codebase': codebase,
+                    'project': project,
+                    'sourcestampsetid': sourcestampsset_id,
+                })
+
+                # Create buildsets
+                res = conn.execute(self.db.model.buildsets.insert(), {
+                    'reason': reason,
+                    'sourcestampsetid': sourcestampsset_id,
+                    'submitted_at': submitted_at,
+                    'complete': bool(complete_at),
+                    'complete_at': complete_at,
+                    'results': results,
+                })
+                buildset_id = res.inserted_primary_key[0]
+
+                # Create buildrequests
+                res = conn.execute(self.db.model.buildrequests.insert(), {
+                    'buildsetid': buildset_id,
+                    'buildername': buildername,
+                    'proiority': 50,
+                    'complete': bool(complete_at),
+                    'results': results,
+                    'submitted_at': submitted_at,
+                    'complete_at': complete_at,
+                    'slavepool': slavepool,
+                })
+                buildrequest_id = res.inserted_primary_key[0]
+
+                # Create builds
+                conn.execute(self.db.model.builds.insert(), {
+                    'number': number,
+                    'brid': buildrequest_id,
+                    'slavename': slavename,
+                    'start_time': submitted_at,
+                    'finish_time': complete_at,
+                })
+                transaction.commit()
+                return True
+            except Exception as e:
+                print("Exception occurs during create new build", e)
+                transaction.rollback()
+                return False
```

`return True | False` looks weird to me in Python.

Why not just re-raise the exception after rollback?

---

Solved 👍